### ISSUE TITLE
Support for non-standard HTTP responses (e.g. ICY)

### ISSUE THREAD

**Issue by storkme** [square/okhttp#386]
Hi,

This isn't exactly a bug per se, but it is causing me (and [this person](http://stackoverflow.com/questions/19738798/android-4-4-http-api-bugs) and [this person](http://stackoverflow.com/questions/19759966/how-to-parse-a-none-standard-http-response)) some issues.

OkHttp is _stricter_ in parsing HTTP status lines than HttpURLConnection is. Specifically this concerns the protocol version returned in the status line. In a standard HTTP request this'll be `HTTP/1.1`, but when connecting to an Icy Shoutcast server it'll return `ICY`, for instance `ICY 200 OK`.

As I mentioned, HttpURLConnection (on pre-4.4) loads this 'fine' (the statuscode is actually always -1 and statusmessage is null but it doesn't throw up an exception and continues on okay), but OkHttp won't parse it at all.

I realise that you have no obligation whatsoever to implement non-standard parts of the Http spec, but given that OkHttp is now the default Http implementation on KitKat devices from my perspective this is kind of a big deal -- right now I'm checking to see if the device is a KitKat device and then using my own hand-rolled Socket instead of the default HttpUrlConnection that works fine on previous versions.

Details on the ICY spec are few and far between, but in any incarnation I've seen it, it behaves exactly like HTTP but returns a different protocol in the status line. I'd imagine if you're trying to do any transparent caching or content negotiation or anything clever like that you'd probably hit a snag, though.

TL;DR: not throwing an exception when the server response anything other than HTTP/\d.\d would be great :)

If you need any details from me please give me a shout.

Thanks!

P.S. huge fan of your work(s)!

**Comment by kenmasumitsu** [comment]
With the patch, Response.protocol() returns Protocol.HTTP_1_0 for Shoutcast. 
So we cannot distinguish whether it is HTTP/1.0 or Shoutcast. 

Could you add API to return raw protocol string ? (e.g. Response.rawProtocolString().  It returns ICY, HTTP/1.0 or HTTP/1.1 ,... )

**Comment by cloewen8** [comment]
A bit late, but ran into this problem today with a different protocol (NTRIP r1 uses SOURCETABLE and ICY). Supporting ICY is great but I would love to see support for custom protocols (not just an exception for ICY).
